pkg/server/server.go

```diff
@@ -282,6 +299,109 @@ func (s *applicationServer) ListCommits(ctx context.Context, msg *pb.ListCommits
 	}, nil
 }
 
+const KustomizeNameKey string = "kustomize.toolkit.fluxcd.io/name"
+const KustomizeNamespaceKey string = "kustomize.toolkit.fluxcd.io/namespace"
+
+func (s *applicationServer) GetReconciledObjects(ctx context.Context, msg *pb.GetReconciledObjectsReq) (*pb.GetReconciledObjectsRes, error) {
+	if msg.AutomationKind == pb.GetReconciledObjectsReq_Helm {
+		return nil, grpcStatus.Error(codes.Unimplemented, "Helm is not currently supported for this method")
+	}
+	result := []unstructured.Unstructured{}
+
+	for _, gvk := range msg.Kinds {
+		list := unstructured.UnstructuredList{}
+
+		list.SetGroupVersionKind(schema.GroupVersionKind{
+			Group:   gvk.Group,
+			Kind:    gvk.Kind,
+			Version: gvk.Version,
+		})
+
+		opts := client.MatchingLabels{
+			KustomizeNameKey:      msg.AutomationName,
+			KustomizeNamespaceKey: msg.AutomationNamespace,
+		}
+
+		if err := s.kube.List(ctx, &list, opts); err != nil {
+			return nil, fmt.Errorf("could not get unstructured list: %s\n", err)
+		}
+
+		result = append(result, list.Items...)
+
+	}
+
+	objects := []*pb.UnstructuredObject{}
+	for _, obj := range result {
+		res, err := status.Compute(&obj)
+
+		if err != nil {
+			return nil, fmt.Errorf("could not get status for %s: %w", obj.GetName(), err)
+		}
+
+		objects = append(objects, &pb.UnstructuredObject{
+			GroupVersionKind: &pb.GroupVersionKind{
+				Group:   obj.GetObjectKind().GroupVersionKind().Group,
+				Version: obj.GetObjectKind().GroupVersionKind().GroupVersion().Version,
+				Kind:    obj.GetKind(),
+			},
+			Name:      obj.GetName(),
+			Namespace: obj.GetNamespace(),
+			Status:    res.Status.String(),
+			Uid:       string(obj.GetUID()),
+		})
+	}
+	return &pb.GetReconciledObjectsRes{Objects: objects}, nil
+}
+
+func (s *applicationServer) GetChildObjects(ctx context.Context, msg *pb.GetChildObjectsReq) (*pb.GetChildObjectsRes, error) {
```

Comment:
This works as it is, but it seems like it would be a nicer abstraction if we maintained the types of child objects separately so that a user of this function could just pass in the parent object without having to specify the child type(s); that would also support an eventual recursive descendant search

Reply:
@jrryjcksn Kubernetes won't support that. An object doesn't know the types of children it has unfortunately.

Reply:
Oh I see what you are saying. We store this in the UI right now. It can be moved to the backend, but we don't know ALL of the parent-child relationships, so it made sense to push this onto the user.

Reply:
In the case of a CRD, maybe its impossible for us to know what the relationships are, which makes me lean towards the current API design of asking the user to specify the objects they care about.